lochness/box/__init__.py

```diff
@@ -65,16 +65,38 @@ def get_box_object_based_on_name(client:boxsdk.client,
 
     Returns:
         box_folder_object
+
     '''
     box_folder_name = str(box_folder_name)
 
+    # if box root is given as path - eg) box_folder_name == 'example/PronetLA'
+    if '/' in box_folder_name:
+        # remove leading '/' from the path string
+        box_folder_name = box_folder_name[1:] if box_folder_name[0] == '/' \
+                else box_folder_name
+
+        # get root object
+        root = Path(box_folder_name).parts[0]
+        box_obj = get_box_object_based_on_name(client,
+                                               root,
+                                               box_path_id)
+
+        if box_path_id == '0':
+            box_obj = get_box_object_based_on_name(
+                    client,
+                    Path(box_folder_name).relative_to(root),
+                    box_obj.id)
+
+        if box_obj is None:
+            return None
+
     # get list of files and directories under the top directory
     root_dir = client.folder(folder_id=box_path_id).get()
 
-    # for entry in listing.entries:
+    # Return matched box object
     for file_or_folder in root_dir.get_items():
         if file_or_folder.type == 'folder' and \
-           file_or_folder.name == box_folder_name:
+           file_or_folder.name == box_folder_name.name:
```

Comment:
The purpose of this modification is not clear to me. In the old code, there was a `box_folder_name` in Ln 69. That variable was used in Ln 77. In the new code, some modification has been attempted between the two lines. But I do not see how the modification allowed you to use `box_folder_name.name` in the new code.

Reply:
So the goal is--if a string `example/ProNET_LA` is provided, you want to use just `ProNET` and omit the part before `/`, right?

Reply:
>but I do not see how the modification allowed you to use box_folder_name.name in the new code.

Good point. `box_folder_name.name` looks like a mistake. Let me test again and fix this.

Reply:
The `box` api does not allow you pull files with a path. It requires ID or `box` object to be specified before downloading a file. So, if the file in interest was saved in a subdirectory, we have to
1. get the id of the root
2. get the id of the directory below root
3. get the id of the directory below the directory found in 2
and so on.

Given your example above, the goal is to get the `box` object of the `ProNET_LA` under `example`.

Reply:
> > but I do not see how the modification allowed you to use box_folder_name.name in the new code.
> 
> Good point. `box_folder_name.name` looks like a mistake. Let me test again and fix this.

Thanks for spotting this. Fixed & tested in https://github.com/PREDICT-DPACC/lochness/pull/31/commits/a1dfc9834c06f695ebd8f8d0c47e9ba3cbf7ac79